test/integration/client/client_test.go

```diff
@@ -795,3 +800,56 @@ func TestSelfLinkOnNamespace(t *testing.T) {
 
 	runSelfLinkTestOnNamespace(t, c, "default")
 }
+
+func TestEventCompatibility(t *testing.T) {
```

Comment:
The test itself looks fine - it's what I wanted to see.

My only concern is that it shouldn't live in this file - we should probably create a dedicated directory:
test/integration/events
and move this test there.